src/ggrc/maintenance_views/maintenance.py

```
@@ -0,0 +1,151 @@
+      else:
+        msg = "Invalid access token"
+        logger.info(msg)
+        return msg
+    else:
+      msg = "Access token not found in settings"
+      logger.info(msg)
+      return msg
+  else:
+    gae_user = users.get_current_user()
+    if gae_user and gae_user.email() in settings.BOOTSTRAP_ADMIN_USERS:
+      run_migration()
+    else:
+      msg = "User not authorized"
+      logger.info(msg)
+      return msg
+    return redirect(url_for('index'))
+
+
+@maintenance_app.route('/maintenance/turnoff_maintenance_mode',
+                       methods=['POST'])
+def turn_off_maintenance_mode():
+  """Super users are allowed to turn off maintenance mode manually."""
+  gae_user = users.get_current_user()
+  if gae_user and gae_user.email() in settings.BOOTSTRAP_ADMIN_USERS:
+    sess = db.session
+    db_row = sess.query(Maintenance).get(1)
+
+    # Set the db flag before running migrations
+    if db_row:
+      db_row.under_maintenance = False
+      sess.add(db_row)
+      sess.commit()
+  else:
+    msg = "User not authorized"
+    logger.info(msg)
+    return msg
+
+  return redirect(url_for('index'))
+
+
+@maintenance_app.route('/maintenance/check_migration_status/<row_id>',
+                       methods=['GET'])
+def check_migration_status(row_id):
+  """Checks and returns the status of migration."""
+  try:
+    sess = db.session
+    maint_row = sess.query(Maintenance).get(1)
+    mig_row = sess.query(MigrationLog).get(row_id)
+    if mig_row:
+      if mig_row.log:
+        return "Fail,{}".format(mig_row.log)
+      elif mig_row.is_migration_complete:
+        return "Complete"
+      elif not maint_row.under_maintenance:
+        return "Fail,{}".format("Migration seem to stuck.")
+      else:
+        return "In progress"
+    else:
+      return "Fail,{}".format("No migration entry in db.")
+  except sqlalchemy.exc.ProgrammingError as e:
+    if not re.search(r"""\(1146, "Table '.+' doesn't exist"\)$""", e.message):
+      raise
```

Is this code just suppressing all table doesn't  exist errors? Shouldn't we at at least log it when that happens?

---

Actually it suppresses these errors because when migration is run for the first time, the maintenance tables are also part of migration and it won't exist for the first time.

But you are right we should at least log it because it might occur due to other table not existing.